service/src/main/java/uk/gov/hmcts/reform/fpl/service/RemoveOrderService.java

```diff
@@ -62,4 +66,36 @@ public void hideOrder(List<Element<GeneratedOrder>> orders, List<Element<Generat
                 hiddenOrders.add(order);
             });
     }
+
+    public List<Element<Child>> removeFinalOrderPropertiesFromChildren(CaseData caseData) {
+        UUID id = getDynamicListValueCode(caseData.getRemovableOrderList(), mapper);
+        List<Element<Child>> children = caseData.getAllChildren();
+
+        findElement(id, caseData.getOrderCollection());
+
+        Optional<Element<GeneratedOrder>> removedOrder = findElement(id, caseData.getOrderCollection());
+
+        if (removedOrder.isEmpty()) {
+            throw new IllegalArgumentException("Failed to find the order to be removed");
```

Comment:
Not so sure IllegalArgumentException is right, @samuelzerozero thoughts?

Reply:
how about IllegalStateException?

Reply:
actually, IllegalArgunment is good